nullaway/src/main/java/com/uber/nullaway/fixserialization/out/ErrorInfo.java

```diff
@@ -22,21 +22,53 @@
 
 package com.uber.nullaway.fixserialization.out;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.util.TreePath;
 import com.uber.nullaway.ErrorMessage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /** Stores information regarding an error which will be reported by NullAway. */
 public class ErrorInfo {
 
   private final ErrorMessage errorMessage;
   private final EnclosingClassAndMethodInfo enclosingInfo;
 
+  /** Special characters that need to be escaped in TSV files. */
+  private static final ImmutableMap<Character, Character> escapes =
+      ImmutableMap.of(
+          '\n', 'n',
+          '\t', 't',
+          '\f', 'f',
+          '\b', 'b',
+          '\r', 'r');
+
   public ErrorInfo(TreePath path, ErrorMessage errorMessage) {
     this.enclosingInfo = new EnclosingClassAndMethodInfo(path);
     this.errorMessage = errorMessage;
   }
 
+  /**
+   * Escapes special characters in string to conform with TSV file formats. The most common
+   * convention for lossless conversion is to escape special characters with a backslash according
+   * to <a
+   * href="https://en.wikipedia.org/wiki/Tab-separated_values#Conventions_for_lossless_conversion_to_TSV">
+   * Conventions for lossless conversion to TSV</a>
+   *
+   * @param str String to process.
+   * @return returns modified str which its special characters are escaped.
+   */
+  private static String escapeSpecialCharacters(String str) {
+    // escape existing backslashes
+    str = str.replaceAll(Pattern.quote("\\"), Matcher.quoteReplacement("\\\\"));
+    // escape special characters
+    for (Character key : escapes.keySet()) {
+      str = str.replaceAll(String.valueOf(key), "\\\\" + escapes.get(key));
```

Comment:
Why four slashes here? Shouldn't we just want to add a single `\` to the actual internal string representation, meaning the encoding on the string literally should just be `\\`? (Above we are re-encoding single slashes as double slashes, but we want e.g. new line replaced by `\n` internally: single-slash, followed by n)

Reply:
It seems to be working based on the test, so maybe I am missing something here. But, in that case we want a comment explaining it.

Reply:
In regular expression `\` is a separate special character from `\n` and `\t` and etc. To match exactly `\` we need to pass "\\\\". However in order to create `\\` with string in java, each backslash must also be escaped, therefore, we need four `\` to make regex match only the character `\`.

24acf2781088486f4f62ef0133d59f7503735101

Reply:
I used `quoteReplacement()` for both cases to make it more readable.

944d6ac53fcc19715c41e6f884aa5ac152fdc142

Reply:
My point is that I don't think the second argument of `replaceAll(...)` is a regex, but a string. In which case "\\\\n" becomes the string for [`\`,`\`,`n`] in memory, where we want [`\`,`n`]. Unless I am missing something.